pyccel/ast/cwrapper.py

```diff
@@ -372,16 +378,34 @@ def args(self):
                         ('complex',8)  : numpy_cdouble_type,
                         ('complex',16) : numpy_clongdouble_type}
 
-def PyType_Check(data_type):
+
+def Type_Check(data, argument):
+    # check for numpy type
+    try :
+        check_numpy_ref = numpy_type_check_registry[(data.dtype, data.precision)]
+    except KeyError:
+        errors.report(PYCCEL_RESTRICTION_TODO, symbol=data.dtype,severity='fatal')
+
+    check_numpy_func = FunctionDef(name = 'PyArray_IsScalar',
+                    body = [],
+                    arguments = [Variable(dtype=PyccelPyObject(), name = 'o', is_pointer=True), check_numpy_ref],
+                    results   = [Variable(dtype=NativeBool(), name = 'r')])
+
+    # check for python type
     try :
-        check_type = check_type_registry[data_type]
+        check_type = python_type_check_registry[data.dtype]
     except KeyError:
-        errors.report(PYCCEL_RESTRICTION_TODO, symbol=data_type,severity='fatal')
-    func = FunctionDef(name = check_type,
+        errors.report(PYCCEL_RESTRICTION_TODO, symbol=data.dtype,severity='fatal')
```

Comment:
yes already tested that's why i created the new check_type function that takes into consideration the numpy types

Reply:
Ah, wait it works because you just check the dtype right? But does that not mean that your check for an `int8` would be:
```
if ( PyArray_IsScalar(arg, Byte) || PyLong_Check(arg) )
```
?
Have you tried passing an `int` to the prototype above?

Reply:
I'm checking for type with precision first in numpy_type_check_registry after that i go create a check for python type 
the numpy one need type and precision  and python on need only the type .

Reply:
But if you use an `or` then only one needs to be right so a long should pass the test when you're expecting an int8 shouldn't it?
I'm not clear on why you need both of these. Maybe we should discuss this again after PR #530 is merged and you have added tests